packages/react-internal/src/components/Modal/Modal.types.ts

```diff
@@ -63,6 +63,12 @@ export interface IModalProps extends React.RefAttributes<HTMLDivElement>, IAcces
    */
   componentRef?: IRefObject<IModal>;
 
+  /**
+   * Optional callback ref to access the modal container. Use this to access the properties
+   * of the modal rectangle container.
+   */
+  containerRef?: (container: HTMLDivElement | null) => void;
```

Comment:
This should probably be the actual ref type rather than a callback.
```suggestion
  containerRef?: React.Ref<HTMLDivElement>;
```